### PR TITLE
[Fix-7291][datasource] support driverClassName default

### PR DESCRIPTION
**Pull request by Narcasserun**
- support driverClassName default
- Compatible with different versions of MySQL data source loading

### CODE REVIEW

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/provider/JdbcDataSourceProvider.java [hunk 1]
```diff
@@ -80,4 +89,23 @@ public static HikariDataSource createOneSessionJdbcDataSource(BaseConnectionPara
...
+        try {
+            final Class<?> clazz = Class.forName(drv, true, classLoader);
+            final Driver driver = (Driver) clazz.newInstance();
+            if (!driver.acceptsURL(properties.getJdbcUrl())) {
+                logger.warn("Jdbc driver loading error. Driver {} cannot accept url.", drv);
+                throw new RuntimeException("Jdbc driver loading error.");
+            }
+            if (dbType.equals(DbType.MYSQL) && driver.getMajorVersion() >= 8) {
+                properties.setDriverClassName(drv);
+            } else {
```
Comment (CalvinKirs) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/provider/JdbcDataSourceProvider.java]: If it is other db, will there be a problem here?

Comment (Narcasserun) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/provider/JdbcDataSourceProvider.java]: yes

**Review by CalvinKirs — APPROVED** [review]
+1

**Review by lenboo — APPROVED** [review]
+1

**Pull request merged**